src/modules/Dashboard/features/Applet/Popups/UpgradeAccountPopup/UpgradeAccountPopup.tsx

```diff
@@ -0,0 +1,148 @@
+import { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useForm } from 'react-hook-form';
+import { yupResolver } from '@hookform/resolvers/yup';
+
+import { Modal, Spinner, SubmitBtnVariant } from 'shared/components';
+import {
+  StyledBodyLarge,
+  StyledErrorText,
+  StyledFlexColumn,
+  StyledModalWrapper,
+} from 'shared/styles';
+import { useFormError } from 'modules/Dashboard/hooks';
+import { Mixpanel, getErrorMessage } from 'shared/utils';
+import { Languages, postSubjectInvitationApi } from 'api';
+import { useAppDispatch } from 'redux/store';
+import { useAsync } from 'shared/hooks';
+import { banners } from 'redux/modules';
+import { AccountType } from 'modules/Dashboard/types';
+import { ParticipantSnippet } from 'modules/Dashboard/components';
+
+import { RESPONDENT_ALREADY_INVITED } from './UpgradeAccountPopup.const';
+import { UpgradeAccountPopupSchema } from './UpgradeAccountPopup.schema';
+import {
+  UpgradeAccountPopupProps,
+  UpgradeAccountFormValues,
+  Fields,
+} from './UpgradeAccountPopup.types';
+import { UpgradeAccountForm } from './UpgradeAccountForm';
+
+export const UpgradeAccountPopup = ({
+  popupVisible,
+  appletId,
+  subjectId,
+  secretId,
+  nickname,
+  tag,
+  onClose,
+  'data-testid': dataTestid,
+}: UpgradeAccountPopupProps) => {
+  const { t, i18n } = useTranslation('app');
+  const defaultValues = {
+    email: '',
+    language: i18n.language as Languages,
+  };
+  const {
+    handleSubmit,
+    control,
+    reset,
+    setError,
+    formState: { isValid, isDirty },
+  } = useForm<UpgradeAccountFormValues>({
+    resolver: yupResolver(UpgradeAccountPopupSchema()),
+    defaultValues,
+    mode: 'all',
+    delayError: 1000,
+  });
+  const dispatch = useAppDispatch();
+  const [hasCommonError, setHasCommonError] = useState(false);
+
+  const handleClose = (shouldRefetch = false) => {
+    resetForm();
+    setHasCommonError(false);
+    onClose?.(shouldRefetch);
+  };
+
+  const resetForm = () => reset(defaultValues);
+
+  const {
+    error,
+    execute: createInvitation,
+    isLoading,
+  } = useAsync(postSubjectInvitationApi, async (result) => {
+    dispatch(
+      banners.actions.addBanner({
+        key: 'AddParticipantSuccessBanner',
+        bannerProps: {
+          accountType: AccountType.Full,
+          id: result.data?.result?.secretUserId,
+        },
+      }),
+    );
+    Mixpanel.track('Invitation sent successfully');
+    handleClose(true);
+  });
+
+  const handleSubmitForm = (values: UpgradeAccountFormValues) => {
+    if (!appletId || !subjectId) return;
+    Mixpanel.track('Subject Invitation click');
+
+    createInvitation({
+      appletId,
+      subjectId,
+      ...values,
+    });
```

Comment:
Only issue I'm seeing is the invite email gets sent in English for both language options. For reference, "Add Participant" sends the email in the correct language.

Reply:
You're right, it's missing from the API 😭 I'll add support for it to the API client callback (which thankfully the BE doesn't mind my doing) so that it's strictly a BE-only change.

Reply:
Created [M2-6283](https://mindlogger.atlassian.net/browse/M2-6283)